frontend/src/components/header/MenuOverlay.js

```diff
@@ -651,6 +655,7 @@ class MenuOverlay extends Component {
                       this.handleKeyDown,
                       DEBOUNCE_TIME_SEARCH
                     )}
+                    onKeyUp={this.setLastQuery}
```

Comment:
what's the reason behind this?

Reply:
i mean, why is not enough to capture the last query exacly when we set the result?

Reply:
have the last query string recorded so it wont be debounced

Reply:
so now it's OK to not have it in redux, right? :)

Reply:
as the documentation that you pointed emphasises it is a balance. For the case of the Table because it is a complex component I chosen that redux update approach. In this case this is more appropriate (note that is not updating the state such that it doesn't cause re-rendering, it's updating local scoped flag)

Reply:
Q: why are u clearing it when u get the results?
IMHO u shall keep it in the state as follows:
* queriedResult - no change here, keep it like it is now
* query - the query which comes with the queriedResult or which is issued when we start to query ..... so there we always have the last query
* pendingQuery - flag to indicate if the `query` is currently running or not

why do we need to that `this.lastQuery` don't understand?
it feels like spaghetti just born...

Reply:
I am setting it to the value of the input on keyUpEvent to have the string used to be queried stored without causing an extra  re-render due to state change. If I were to set this in the state using setState most probably there will be also times when it will be updated in the state after the completion of the query. Note that setState is async and wanted to avoid this. Now the issue that I am having is with the debouncing.

Reply:
<img width="1636" alt="Screenshot 2022-01-25 at 14 56 23" src="https://user-images.githubusercontent.com/1708561/150980917-dc1f1a47-1720-4877-a562-a06afcb9ed64.png">
Somehow the logic I want to have does not work as I will have all the time equality between the string from the input (e.target.value and the `this.queryString`)

Reply:
I think we also need to increase the debounce time